user.py

```diff
@@ -1,15 +1,19 @@
 from decorators import login_required
-from db import db, User
-from flask import Blueprint, render_template, flash, current_app, url_for, request, make_response, redirect, session
-import re
+from db import db, User, Researcher, Course, PreferenceAssignment, Teacher
+from flask import Blueprint, render_template, flash, url_for, request, make_response, redirect, session, \
+    Flask
+from enums import DEFAULT_MAX_LOAD
+from util import get_current_year
+import re, json
 
 user_bp = Blueprint('user', __name__)
 
 
 @user_bp.route('/register')
 @login_required
 def register():
-    return render_template('register.html')
+    supervisors = db.session.query(User).filter(User.admin == False, User.is_teacher == True, User.active == True).all()
```

Comment:
Is there any reason to filter out admins from the potential supervisors ?

Reply:
No particular reason, I didn't take into account the case where a teacher is an admin.